src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java

```diff
@@ -79,20 +80,18 @@ private void execute( final Consumer<RecheckLifecycle> lifecycleMethod, final Ob
 	}
 
 	private boolean isRecheckLifecycle( final Field field, final Object testInstance ) {
-		final boolean accessibility = unlock( field );
-		try {
-			return RecheckLifecycle.class.isInstance( field.get( testInstance ) );
-		} catch ( IllegalArgumentException | IllegalAccessException e ) {
-			throw new IllegalStateException( e );
-		} finally {
-			lock( field, accessibility );
-		}
+		return applyOnFieldValue( field, testInstance, value -> value instanceof RecheckLifecycle );
 	}
 
 	private RecheckLifecycle getRecheckLifecycle( final Field field, final Object testInstance ) {
+		return applyOnFieldValue( field, testInstance, value -> (RecheckLifecycle) value );
+	}
+
+	private <T> T applyOnFieldValue( final Field field, final Object testInstance, final Function<Object, T> f ) {

```

Comment:
Nice one. Didn't think about that when implementing the extension. Could we find a better name for `f`?

Reply:
`fun` (like in Kotlin), `func`, `function`, `toApply`, … ?

Reply:
What about `converter`?
Otherwise I would vote for `toApply`.

Reply:
If I rename the method to `mapFieldValue` and then name the function param `mapper`? But we can also ge for `convertFieldValue` and `converter`.

Reply:
I prefer the `map` variant, as it is more common in processing `Stream`s and `Optional`s.